docs/shared/tutorials/react-standalone.md

```diff
@@ -392,22 +479,133 @@ Nx allows you to separate this logic into "local libraries". The main benefits i
+Let's assume our domain areas include `products`, `orders` and some more generic design system components, called `ui`. We can generate a new library for these areas using the React library generator:
 
 ```
 nx g @nx/react:library products --unitTestRunner=vitest --bundler=none --directory=modules/products
+```
+
+Note how we use the `--directory` flag to place the library into a subfolder. You can choose whatever folder structure you like to organize your libraries.
+
+Nx tries to set up your workspace to work with the modular library architecture, but depending on your existing configuration, you may need to tweak some settings. In this repo, you'll need to do a few things in order to prepare for future steps.
+
+#### Lint Settings
+
+Install the `@nx/eslint-plugin` package:
+
+```
+npx nx add @nx/eslint-plugin
+```
+
+We want the `lint` task for the root `react-app` project to only lint the files for that project, so we'll change the `lint` command in `package.json`:
+
+```json {% fileName="package.json" %}
+{
+  "scripts": {
+    "lint": "eslint src --ext ts,tsx --report-unused-disable-directives --max-warnings 0"
+  }
+}
+```
+
+Now we need to update the `.eslintrc.cjs` file to extend the `.eslintrc.base.json` file:
+
+```js {% fileName=".eslintrc.cjs" highlightLines=[8,10] %}
+module.exports = {
+  root: true,
+  env: { browser: true, es2020: true },
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:react-hooks/recommended',
+    './.eslintrc.base.json',
+  ],
+  ignorePatterns: ['!**/*', 'dist', '.eslintrc.cjs'],
+  parser: '@typescript-eslint/parser',
+  plugins: ['react-refresh'],
+  rules: {
+    'react-refresh/only-export-components': [
+      'warn',
+      { allowConstantExport: true },
+    ],
+  },
+};
+```
+
+We'll also set `enforceBuildableLibDependency` to `false` in `.eslintrc.base.json`.
+
+```{% fileName=".eslintrc.base.json" %}
+  "enforceBuildableLibDependency": false,
+```
+
+#### Build Settings
+
+To make sure that the build can correctly pull in code from libraries, we'll update `tsconfig.base.json` and `vite.config.ts`.
+
+```json {% fileName="tsconfig.base.json" highlightLines=[9] %}
+{
+  "compileOnSave": false,
+  "compilerOptions": {
+    "moduleResolution": "bundler",
+    "target": "ES2020",
+    "module": "ESNext",
+    "lib": ["ES2020", "DOM", "DOM.Iterable"],
+    "skipLibCheck": true,
+    "baseUrl": ".",
+    "paths": {
+      "products": ["modules/products/src/index.ts"]
+    }
+  }
+}
+```
+
+```js {% fileName="vite.config.ts" highlightLines=[3,7] %}
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react(), nxViteTsPaths()],
```

Comment:
Same with the tsconfig paths for Vite.

Reply:
I'm working on it so this would go away. This would be for both Vue and React tutorials.

Reply:
In 19.3.0 you can run `nx g @nx/vite:setup-paths-plugin`, or use `nx add @nx/vite --setupPathsPlugin`.